bots/psuteam7bot/movement.js

```diff
@@ -0,0 +1,408 @@
+    }
+    if(quadrant === 3)  //Enemy in quadrant 1 or 4
+    {
+        Ax = (x+distX);
+        By = (y-distY);
+    }
+    if(quadrant === 4)  //Enemy in quadrant 2 or 3
+    {
+        Ax = (x-distX);
+        By = (y-distY);
+    }
+
+    return [{x: Ax, y}, {x, y: By}];
+}
+
+/*Check and return whether tile at specified coordinate is passable
+*Used for movement, placing built/ spawned units
+*Input:     location    -   the robot's 'position/ location' object, should be self.me
+*           fullMap     -   the full map, should be self.map or self.getPassableMap()
+*           robotMap    -   robot map, should be self.getVisibleRobotMap()
+
+*Output:    RetVal  -   true if 'location' is passable, false otherwise
+*/
+movement.isPassable = (location, fullMap, robotMap) => {
+    const {x, y} = location;
+
+    if(x < 0 || y < 0)  //Map bound check
+        return false;
+    else if(x > fullMap.length || y > fullMap.length)   //Map bound check
+        return false;
+    return((robotMap[y][x] === 0) && (fullMap[y][x])); //Returns true only if tile is empty and is passable
+}
+
+/*Return an array of resource depot locations sorted by distance from location passed as parameter
+*Use Case: for pioneers?
+*Might exceed chess clock? remove if so...
+*TODO: Might be unnecessary
+*/
+/*
+movement.getSortedResourceList(location, resourceMap)
+{
+    const length = resourceMap.length;
+    var sortedArr = [];
+    var distArr = [];
+    var currentDist;
+
+    for (let y = 0; y < length; ++y) 
+    {
+        for (let x = 0; x < length; ++x) 
+        {
+            if (resourceMap[y][x]) 
+            {
+                currentDist = movement.getDistance(location, {x, y});
+
+                if(sortedArr.length === 0)
+                {
+                    sortedArr = [{x, y}];
+                    distArr = [currentDist];
+                }
+
+                for(let i = 0; i < sortedArr.length; ++i)
+                {
+                    if(currentDist < distArr[i])
+                    {
+                        sortedArr.splice(i, 0, {x, y});
+                        distArr.splice(i, 0, currentDist);
+                        break;
+                    }
+                }
+            }
+        }
+    }
+    return sortedArr;
+}
+*/
+
+/*A more simple moveTowards, move to a nearby passable adjacent tile, hopefully closer to destination
+*Input: self        -   The robot unit
+*       destination -   The destination 'position/ location' object {x, y}
+*Output:    retVal  -   A 'position/ location' object {x, y} which is adjacent to self.me 
+*                   -   OR self.me if they are all not passable
+*version of dumberMoveTowards with robot not recording a previous, might get stuck in corner
+*/
+/*
+movement.dumberMoveTowards = (self, destination) => {
+    let fullMap = self.map;
+    let robotMap = self.getVisibleRobotMap();
+    let direction = getRelativeDirection(self.me, destination);
+    let {x, y} = self.me;
```

Comment:
Can I clarify what this is doing? Is it setting both `x` and `y` equal to `self.me`, getting the values of `self.me.x` and `self.me.y`, or something else? I get this is commented out, but just want to double check the process for this in case it's used elsewhere